src/dfmt/formatter.d

```diff
@@ -1401,12 +1401,14 @@ private:
         case tok!".":
             regenLineBreakHintsIfNecessary(index);
             immutable bool ufcsWrap = astInformation.ufcsHintLocations.canFindIndex(current.index);
```

Comment:
I was able to get your tests to pass by reverting all these changes and replacing them with
```d
immutable bool ufcsWrap = config.dfmt_keep_line_breaks == OptionalBoolean.f
        && astInformation.ufcsHintLocations.canFindIndex(current.index);
```

Reply:
I wasn't aware I broke long line wrapping. I applied your suggestion and added a test for a long line. dfmt still breaks the line at a strange position: not 80 characters, not 120, but 100. But I think it is a different thing.